File: Employee.cpp
#include "Employee.h"
#include "Utils.h"


int ShpaginEmployee::max_id = 0;

void ShpaginEmployee::reset_max_id()
{
	max_id = 0;
}

ShpaginEmployee::ShpaginEmployee()
{
	id = ++max_id;
	name = L"";
	age = 0;
	experience = 0;
	salary = 0;
}

void ShpaginEmployee::console_input()
{
    std::wcout << L"фамилия и имя: ";
	std::getline(std::wcin >> std::ws, name);
	age = GetCorrectNumber(std::cin, 18, 100, L"возраст: ");
	experience = GetCorrectNumber(std::cin, 0, age, L"стаж работы: ");
	salary = GetCorrectNumber(std::cin, 0, INT_MAX, L"зарпата (в месяц): ");
}

void ShpaginEmployee::console_output() const
{
	std::wcout << L"Сотрудник id_" << id << std::endl
			   << L"имя - " << name << std::endl
			   << L"возраст - " << age << std::endl
			   << L"стаж работы - " << experience << L" год/года/лет" << std::endl
			   << L"зарплата - " << salary << L" руб/мес" << std::endl;
}

File: CSharpWithForms/Employee.cs
namespace CSharpWithForms
{
    public class ShpaginEmployee
    {
        public static int MaxId { get; set; }
        public int Id { get; set; }
        public string Name { get; set; }
        public int Age { get; set; }
        public int Experience { get; set; }
        public int Salary { get; set; }
        public static void ResetMaxId() { MaxId = 0; }

        public ShpaginEmployee()
        {
            Id = ++MaxId;
            Name = "";
            Age = 18;
            Experience = 0;
            Salary = 0;
        }

        public ShpaginEmployee(ShpaginEmployee employee)
        {
            Id = employee.Id;
            Name = employee.Name;
            Age = employee.Age;
            Experience = employee.Experience;
            Salary = employee.Salary;
        }

        public string GetNameForList() => $"#{Id} - {Name}";

    }
}
